### DISCUSSION TITLE
how can you deploy a contract with a linked library?

### DISCUSSION THREAD

**Discussion by oveddan** [wevm/viem#1031] [Question]
Given you have a library with an external function, and you have a contract that calls that external function, how can you deploy these together using their bytecode and `deployContract`?  

For example, you have the following code:

```solidity
library SomeLib {
    function doSomething(uint256 value) external pure {
    }
}

contract MyContract {
   function doSomethingThatCallsLib(uint256 value) external view {
      return SomeLib.doSomething(value);
   }
}
```

I want to be able to deploy `MyContract` using its bytecode:

```ts
await walletClient.deployContract({
   abi: myContractAbi,
   bytecode: myContractByteCode,
   account: deployerAccount
});
```

But this results in an error:  `Invalid parameters were provided to the RPC method.
Double check you have provided the correct parameters.`

I believe we need to configure the linked library address, how can this be done?

**Comment by oveddan** [comment]
full example here: [oveddan/wagmi-external-lib-example](https://github.com/oveddan/wagmi-external-lib-example)

**Comment by pistomat** [comment]
This is a feature I would very much appreciate.

**Comment by Yuripetusko** [comment]
As a workaround, what worked for me was to use bytecode that I copy from foundry artifacts instead of the generated by wagmi.

https://github.com/rmrk-team/rmrk-composable-nft-creator/blob/8077b9d88d0ed04ed4eb32afdc7f111bc8257ba3/package.json#L14

And then deploying it this way

https://github.com/rmrk-team/rmrk-composable-nft-creator/blob/8077b9d88d0ed04ed4eb32afdc7f111bc8257ba3/app/catalog/page.tsx#L61
